src/plugins/data/common/search/search_source/fetch/get_search_params.ts

```diff
@@ -60,9 +60,11 @@ export function getExternalSearchParamsFromRequest(
   const searchParams = getSearchParams(getConfig);
   const dataFrame = getDataFrame();
   const indexTitle = searchRequest.index.title || searchRequest.index;
+  const dataSource = searchRequest.dataSource;
 
   return {
     index: indexTitle,
+    dataSource,
```

Comment:
within the dataframe, dataframe.name should be the datasource. at the time it was implemented it was the index or index pattern.

does the name still work here? if not do you think we can add the data source information to the dataframe.meta field? 

for the async query @sejli is adding query meta infromation like the session id to the df.meta. I think that fits. It also will help with future work with stacking queries as we would want to get the data source attached to query which we will get from the meta field

Reply:
this we wouldn't need to update these parts.

Reply:
agreed but since the dataframe isn't generated by the data plugin initially, it would have to go through the search request for now. might need to be changed when stacked queries are designed and implemented

Reply:
i think we got something working with sean's stuff for this: https://github.com/sejli/OpenSearch-Dashboards/tree/async-polling-3 so we can unite the logic post merge.